keras_nlp/models/gpt_neo_x/rotary_embedding.py

```diff
@@ -11,49 +11,53 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import tensorflow as tf
-
 from keras_nlp.backend import keras
+from keras_nlp.backend import ops
 
 
 class RotaryEmbedding(keras.layers.Layer):
     def __init__(self, max_wavelength=10000, **kwargs):
         super().__init__(**kwargs)
         self.max_wavelength = max_wavelength
 
-    def _apply_rotary_pos_emb(self, tensor, cos_emb, sin_emb):
-        cos_emb = cos_emb[:, : tf.shape(tensor)[1], :, :]
-        sin_emb = sin_emb[:, : tf.shape(tensor)[1], :, :]
+    def build(self, rotary_dim):
+        self.rotary_dim = rotary_dim
+        self.built = True
 
-        x1, x2 = tf.split(tensor, 2, axis=-1)
-        half_rot_tensor = tf.concat((-x2, x1), axis=-1)
+    def __call__(self, query, key):
+        if not self.built:
+            rotary_dim = query.shape[-1]
+            self.build(rotary_dim)
+        return super().__call__(query, key)
 
-        return (tensor * cos_emb) + (half_rot_tensor * sin_emb)
+    def call(self, query, key):
```

Comment:
We should not need this workaround here. Just read `rotary_dim` from `ops.shape(query)[-1]`, we should not need a `build` method at all.

Reply:
It magically works this time, wasn't working earlier LoL

Reply:
> It magically works this time, wasn't working earlier LoL

@mattdangerw it appeared again. Weird thing is I'm not getting this error on local, but CI is reflecting it.